### PR TITLE
Feature/abstract normalization

### PR DESCRIPTION
**Pull request by mottljan**
Adds a new normalization support to be able to fully customize it per project needs via the library itself. Motivation was a crash on one project, where there was a non-ASCII character being passed to the OkHttp header, which is not allowed by it. You can read more in the README or docs

### CODE REVIEW

File: useragent/src/main/java/io/github/ackeecz/useragent/UserAgent.java [hunk 1]
```diff
@@ -1,28 +1,35 @@
...
+
+import io.github.ackeecz.useragent.clientinfo.AndroidClientInfo;
+import io.github.ackeecz.useragent.normalizer.DiacriticsNormalizer;
+import io.github.ackeecz.useragent.normalizer.Normalizer;
 
 /**
- * Singleton class that generates user agent string with app specific data for easier server device distinction
+ * Class that generates user agent string with app specific data for easier server device distinction.
  */
 public class UserAgent {
```
Comment (davidbilik) [inline useragent/src/main/java/io/github/ackeecz/useragent/UserAgent.java]: Why don't you convert this class to kotlin if the rest is in kotlin anyway? For backward compatibility? You could still probably achieve it with kotlin + some java interop annotations, right?

Comment (mottljan) [reply useragent/src/main/java/io/github/ackeecz/useragent/UserAgent.java]: I was too lazy to deal with it, but ok :D f0adf6585f01b8c7e5642f57cf1375cfaf315ac1

**Comment by mishackee** [comment]
LGTM

**Review by mishackee — APPROVED** [review]
LGTM

**Pull request merged**